examples/basics/gloo/spatial_filters.py

```diff
@@ -0,0 +1,163 @@
+uniform float     u_interpolation;
+varying vec2      v_texcoord;
+void main()
+{
+    // using if-statements here is inefficient
+    // used here just for selection of interpolation function
+    if (u_interpolation < 0.5) {
+        gl_FragColor = Nearest(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 1.5) {
+        gl_FragColor = Bilinear(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 2.5) {
+        gl_FragColor = Hanning(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 3.5) {
+        gl_FragColor = Hamming(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 4.5) {
+        gl_FragColor = Hermite(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 5.5) {
+        gl_FragColor = Kaiser(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 6.5) {
+        gl_FragColor = Quadric(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 7.5) {
+        gl_FragColor = Bicubic(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 8.5) {
+        gl_FragColor = CatRom(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 9.5) {
+        gl_FragColor = Mitchell(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 10.5) {
+        gl_FragColor = Spline16(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 11.5) {
+        gl_FragColor = Spline36(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 12.5) {
+        gl_FragColor = Gaussian(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 13.5) {
+        gl_FragColor = Bessel(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 14.5) {
+        gl_FragColor = Sinc(u_texture, u_shape, v_texcoord);
+    }
+    else if (u_interpolation < 15.5) {
+        gl_FragColor = Lanczos(u_texture, u_shape, v_texcoord);
+    }
+    else gl_FragColor = Blackman(u_texture, u_shape, v_texcoord);
+}
+
+"""
+
+
+class Canvas(app.Canvas):
+
+    def __init__(self):
+        app.Canvas.__init__(self, keys='interactive', size=((512), (512)))
+
+        self.program = gloo.Program(VERT_SHADER, FRAG_SHADER)
+        self.texture = gloo.Texture2D(I, interpolation='nearest')
+
+        # using internalformat 'r16f' as discussed in issue #1017
+        self.kernel = gloo.Texture2D(kernel, interpolation='nearest',
+                                     internalformat='r16f')
+
+        self.program['u_texture'] = self.texture
+        self.program['u_kernel'] = self.kernel
+        self.program['u_shape'] = I.shape[1], I.shape[0]
+        self.program['u_interpolation'] = 0
+
+        self.names = [
+            "Nearest", "Bilinear", "Hanning", "Hamming",
+            "Hermite", "Kaiser", "Quadric", "Bicubic",
+            "CatRom", "Mitchell", "Spline16", "Spline36",
+            "Gaussian", "Bessel", "Sinc", "Lanczos", "Blackman"]
```

Comment:
I'm not quite sure I understand your point...

My point is that the `ndarray` you get from `load_spatial_filters.py` can be used with 16 different filters. When loading it, it would be useful to have that list provided, as:
```
kernels, names = load_spatial_filters()
```
where `kernels` is an `ndarray`, and `names` is a tuple of the names that can be used in `glsl` code. Make sense?